Sources/SpeziLocalLLM/LLMLlama+Helpers.swift

```diff
@@ -0,0 +1,77 @@
+//
+// This source file is part of the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import llama
+
+
+/// Extension of ``LLMLlama`` handling the text tokenization.
+extension LLMLlama {
+    /// Converts a textual `String` to the individual `LLMLlamaToken`'s based on the model's dictionary.
+    /// This is a required tasks as LLMs internally processes tokens.
+    ///
+    /// - Parameters:
+    ///   - text: The input `String` that should be tokenized.
+    ///   
+    /// - Returns: The tokenized `String` as `LLMLlamaToken`'s.
+    func tokenize(text: String) -> [LLMLlamaToken] {
+        let nTokens = text.count + (self.parameters.addBos ? 1 : 0)
+        let tokens = UnsafeMutablePointer<llama_token>.allocate(capacity: nTokens)
+        let tokenCount = llama_tokenize(self.model, text, Int32(text.count), tokens, Int32(nTokens), self.parameters.addBos, false)
+        var swiftTokens: [llama_token] = []
+        for tokenIndex in 0 ..< tokenCount {
+            swiftTokens.append(tokens[Int(tokenIndex)])
+        }
+        tokens.deallocate()
+        return swiftTokens
+    }
+    
+    /// Converts a `LLMLlamaToken` to the textual `String` based on the model's dictionary.
+    /// This is a required tasks as LLMs internally processes tokens.
+    ///
+    /// - Parameters:
+    ///   - token: The `LLMLlamaToken` that should be converted.
+    ///   - buffer: A buffer helping with the conversion.
+    ///
+    /// - Returns: The textual `String` of the `LLMLlamaToken`.
+    func token_to_piece(token: LLMLlamaToken, buffer: inout [CChar]) -> String? {
```

Comment:
I would suggest that we keep Swift interfaces using camelCase:
```suggestion
    func tokenToPiece(token: LLMLlamaToken, buffer: inout [CChar]) -> String? {
```

Reply:
Ah sure, was an oversight of me 😄